### PR TITLE
Show a description when suggesting names

### PR DESCRIPTION
**Pull request by rchande**
Fixes https://github.com/dotnet/roslyn/issues/19200

**Customer scenario**

Customer is typing a name and intellisense suggests names. There's no description in the tooltip that is shown on those completion items.

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/19200

**Workarounds, if any**

None

**Risk**

Very low, we simply add a resource string to the constructed completion items

**Performance impact**

Very low, we're passing a string resource into a constructor.

**Is this a regression from a previous update?**

N/A

**Root cause analysis:**

New feature, didn't have a design for the tooltip.

**How was the bug found?**

Dogfooding

### CODE REVIEW

**Comment by rchande** [comment]
This is what the description looks like:
image
Tag @dotnet/roslyn-ide  for review

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs [hunk 1]
```diff
@@ -36,7 +37,7 @@ public class MyClass
     MyClass $$
 }
 ";
-            await VerifyItemExistsAsync(markup, "MyClass", glyph: (int)Glyph.MethodPublic);
+            await VerifyItemExistsAsync(markup, "MyClass", glyph: (int)Glyph.MethodPublic, expectedDescriptionOrNull: CSharpFeaturesResources.suggested_name);
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs]: ❓ Why this line? I expected the following line `myClass`.

File: src/Features/CSharp/Portable/CSharpFeaturesResources.resx [hunk 2]
```diff
@@ -521,4 +521,7 @@
   <data name="Autoselect_disabled_due_to_member_declaration" xml:space="preserve">
     <value>Autoselect disabled due to member declaration</value>
   </data>
+  <data name="suggested_name" xml:space="preserve">
```
Comment (dpoeschl) [inline src/Features/CSharp/Portable/CSharpFeaturesResources.resx]: capitalization of the resource name usually matches the capitalization of the string.

**Comment by dpoeschl** [comment]
Oh, the test didn't fail. What?

**Comment by rchande** [comment]
retest windows_release_vs-integration_prtest

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs (hunk 1 above)
Comment (rchande) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs]: Every item has the same description so I wanted to just update one VerifyItem call. I've made this less ambiguous by adding a separate test that checks the description.

**Comment by rchande** [comment]
@dpoeschl @sharwell The test didn't fail because it is correct (though now made less ambiguous by https://github.com/dotnet/roslyn/pull/19665#discussion_r117578597). Dismissing Sam's review.

**Comment by dpoeschl** [comment]
Thanks for separating that test out. It's way easier to understand what you're testing.

**Comment by rchande** [comment]
test internal-perf please

**Pull request merged**